singularity-eos/eos/eos_gruneisen.cpp

```diff
@@ -127,11 +270,24 @@ PORTABLE_FUNCTION void Gruneisen::DensityEnergyFromPressureTemperature(
     rho = rhom;
   }
 }
-PORTABLE_FUNCTION void Gruneisen::FillEos(Real &rho, Real &temp, Real &sie, Real &press,
+PORTABLE_FUNCTION void Gruneisen::FillEos(Real &rho_in, Real &temp, Real &sie, Real &press,
                                           Real &cv, Real &bmod,
                                           const unsigned long output,
                                           Real *lambda) const {
   // The following could be sped up with work!
+  const unsigned long input = ~output;
+  if (thermalqs::temperature & input && thermalqs::pressure & input) {
+    DensityEnergyFromPressureTemperature(press, temp, lambda, rho_in, sie);
+  }
+  else if (thermalqs::density & output || thermalqs::specific_internal_energy & output) {
+    // Error out on density or energy output because they're currently required as inputs
+    std::stringstream errorMessage;
+    errorMessage << "Gruneisen FillEos: Density and energy are currently required inputs "
+                 << "except when pressure and temperature are inputs"
+                 << std::endl;
+    EOS_ERROR(errorMessage.str().c_str());
+  }
+  const Real rho = std::min(rho_in, _rho_max);
```

Comment:
@jonahm-LANL note that I needed to modify the `FillEos` function a bit to add the density check. You may want to look at this as well.

This is related to issue https://github.com/lanl/singularity-eos/issues/72